ui/src/components/chat/sidebar.rs

```diff
@@ -86,28 +91,52 @@ pub fn Sidebar(cx: Scope<Props>) -> Element {
         state.write().mutate(Action::ChatWith(&chat, true));
     }
 
-    let ch = use_coroutine(cx, |mut rx: UnboundedReceiver<DID>| {
+    let ch = use_coroutine(cx, |mut rx: UnboundedReceiver<MessagesCommand>| {
         to_owned![chat_with];
         async move {
             let warp_cmd_tx = WARP_CMD_CH.tx.clone();
-            while let Some(recipient) = rx.next().await {
-                // if not, create the chat
-                let (tx, rx) = oneshot::channel();
-                if let Err(e) = warp_cmd_tx.send(WarpCmd::RayGun(RayGunCmd::CreateConversation {
-                    recipient,
-                    rsp: tx,
-                })) {
-                    log::error!("failed to send warp command: {}", e);
-                    continue;
-                }
+            while let Some(cmd) = rx.next().await {
+                match cmd {
+                    MessagesCommand::CreateConversation { recipient } => {
+                        // if not, create the chat
+                        let (tx, rx) = oneshot::channel();
+                        if let Err(e) =
+                            warp_cmd_tx.send(WarpCmd::RayGun(RayGunCmd::CreateConversation {
+                                recipient,
+                                rsp: tx,
+                            }))
+                        {
+                            log::error!("failed to send warp command: {}", e);
+                            continue;
+                        }
+
+                        let rsp = rx.await.expect("command canceled");
+
+                        match rsp {
+                            Ok(c) => chat_with.set(Some(c)),
+                            Err(e) => {
+                                log::error!("failed to create conversation: {}", e);
+                                continue;
+                            }
+                        };
+                    }
+                    MessagesCommand::DeleteConversation { conv_id } => {
+                        let (tx, rx) = futures::channel::oneshot::channel();
 
-                let rsp = rx.await.expect("command canceled");
+                        if let Err(e) =
+                            warp_cmd_tx.send(WarpCmd::RayGun(RayGunCmd::DeleteConversation {
+                                conv_id: conv_id,
+                                rsp: tx,
+                            }))
+                        {
+                            log::error!("failed to send warp command: {}", e);
+                            continue;
+                        }
 
-                match rsp {
-                    Ok(c) => chat_with.set(Some(c)),
-                    Err(e) => {
-                        log::error!("failed to create conversation: {}", e);
-                        continue;
+                        let res = rx.await.expect("command canceled");
+                        if res.is_err() {
+                            // failed to delete conversation
```

Comment:
may want to add logging here, but it's almost too minor to mention. This is also fine as is.